### PR TITLE
refactor: improvements

### PR DESCRIPTION
**Pull request by evilrobot-01**
Various fixes and improvements.

### CODE REVIEW

File: crates/pop-parachains/src/call/mod.rs [hunk 1]
```diff
@@ -72,9 +72,9 @@ pub async fn sign_and_submit_extrinsic(
 /// * `tx` - The transaction whose call data will be encoded and returned.
```
Comment (AlexD10S) [inline crates/pop-parachains/src/call/mod.rs:72]:
```suggestion
/// * `xt` - The extrinsic whose call data will be encoded and returned.
```

**Review by AlexD10S — CHANGES_REQUESTED** [review]
Looks good! Just a few minor nitpicks to address, and we’ll be ready to merge.

File: crates/pop-cli/src/commands/call/parachain.rs [hunk 2]
```diff
@@ -235,13 +235,12 @@ impl CallParachainCommand {
...
 		let spinner = cliclack::spinner();
-		spinner.start("Signing and submitting the extrinsic, please wait...");
+		spinner.start("Signing and submitting the extrinsic and then waiting for finalization, please be patient...");
 		let call_data_bytes =
 			decode_call_data(call_data).map_err(|err| anyhow!("{}", format!("{err:?}")))?;
-		let result =
-			sign_and_submit_extrinsic_with_call_data(client.clone(), call_data_bytes, suri)
-				.await
-				.map_err(|err| anyhow!("{}", format!("{err:?}")))?;
+		let result = sign_and_submit_extrinsic_with_call_data(&client, call_data_bytes, suri)
```
[the remark this answers is not in the document]
Comment (evilrobot-01) [reply crates/pop-cli/src/commands/call/parachain.rs]: Addressed in 5693e2fe

**Review by AlexD10S — APPROVED** [review]
LGTM!

**Pull request merged**